sssd_test_framework/utils/sssd.py

```diff
@@ -307,15 +313,26 @@ def set_service_user(self, user: str) -> None:
 
         :param user: Option value to set.
         :type user: str
-        :raises ValueError: If required feature wasn't built.
+        :raises ValueError: in case error happens.
         """
         if isinstance(self.host, BaseHost):
-            if (user != "root") and (not self.host.features["non-privileged"]):
-                raise ValueError("SSSD was built without support of running under non-root")
+            if not self.host.features["non-privileged"]:
+                return
         else:
             raise ValueError("Unexpected host type")
 
-        self.sssd["user"] = user
+        service_file = "/usr/lib/systemd/system/sssd.service"
```

Comment:
Please, add also `self.fs.backup(service_file)` to make sure the file is restored after the test is finished.

Reply:
Done.